discovery/azure/azure.go

```diff
@@ -420,10 +420,20 @@ func (d *Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
 						} else {
 							ch <- target{labelSet: nil, err: err}
 						}
-						// Get out of this routine because we cannot continue without a network interface.
-						return
+						d.addToCache(nicID, networkInterface)
+					} else {
+						networkInterface, err = client.getVMScaleSetVMNetworkInterfaceByID(ctx, nicID, vm.ScaleSet, vm.InstanceID)
```

Comment:
By the way this code is stale/unnecessary, see https://github.com/prometheus/prometheus/pull/13702#discussion_r1514166983
To avoid these merges with a lot of changes,  how about making sure that code changes always go through main first?
So in merges like this we can also expect the changelog, version file etc to change?

cc @bwplotka